scripts/check-i18n-properties.sh

```
@@ -0,0 +1,302 @@
+
+encoding_clean=true
+all_files=("$ENGLISH_FILE" "${LOCALE_FILES[@]}")
+
+for props_file in "${all_files[@]}"; do
+    locale=$(locale_of "$props_file")
+    non_ascii=$(count_non_ascii_lines "$props_file")
+    if [ "$non_ascii" -gt 0 ]; then
+        printf " %-20s  ENCODING VIOLATION: %d line(s) with raw non-ASCII bytes\n" \
+            "${locale}:" "$non_ascii"
+        echo "   Fix: Replace raw chars with \\uXXXX escapes (e.g. é → \\u00e9)"
+        encoding_clean=false
```

Shellcheck (0.11.0)

[warning] encoding_clean appears unused. Verify use (or export if used externally).

(SC2034)